src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -148,7 +149,7 @@ public AzureAppConfigurationOptions Select(string keyFilter, string labelFilter
                 throw new ArgumentException("The characters '*' and ',' are not supported in label filters.", nameof(labelFilter));
             }
 
-            if (!_kvSelectors.Any(s => s.KeyFilter.Equals(keyFilter) && s.LabelFilter.Equals(labelFilter)))
+            if (!_kvSelectors.Any(s => !string.IsNullOrEmpty(s.KeyFilter) && s.KeyFilter.Equals(keyFilter) && !string.IsNullOrEmpty(s.LabelFilter) && s.LabelFilter.Equals(labelFilter)))
```

Comment:
I'm curious about the addition of the nullorempty checks for key filter and label filter. Why are they needed, and also, it doesn't look like it's possible for them to be null or empty given the parameter checks above.

Reply:
Ah, I see because it's used for snapshot now. Perhaps it would be better to check for snapshot null or empty

Reply:
They can be null if SelectSnapshot is called before a Select. I was originally setting keyfilter and labelfilter to a default value but we changed it to leaving it null because we decided it made more sense.

Reply:
Would it be okay to write it as

if (!_kvSelectors.Any(s => string.Equals(s.KeyFilter, keyFilter) &&  string.Equals(s.LabelFilter, labelFilter)))